Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceCompliancePolicyAndroid/MSFT_IntuneDeviceCompliancePolicyAndroid.psm1

```diff
@@ -730,6 +730,11 @@ function Test-TargetResource
     $ValuesToCheck.Remove('TenantId') | Out-Null
     $ValuesToCheck.Remove('ApplicationSecret') | Out-Null
 
+    if($CurrentValues.Ensure -ne $PSBoundParameters.Ensure)
```

Comment:
I just test ensure before the CimInstance to prevent any conflict as when Get-TargetResource return $nullResult the cimInstance (Assignments) hasn't been transformed into hashtable.